test/epoll-test.c

```diff
@@ -42,6 +42,11 @@
 #define EPOLLRDHUP 0x2000
 #endif
 
+/* macOS and Solaris do not have it */
+#ifndef MSG_NOSIGNAL
+#define MSG_NOSIGNAL 0
```

Comment:
`MSG_NOSIGNAL` should be defined as [`SO_NOSIGPIPE`](https://lists.apple.com/archives/macnetworkprog/2002/Dec/msg00091.html).

Reply:
How sure are you? I have seen a fallback to 0 used elsewhere, like https://github.com/lpeterse/haskell-socket/commit/0aa0db55d027bd98d3c9a64ddbd0b230366081d5

Reply:
The suggested behavior is better for macOS, but yeah, it needs to be wrapped in `#ifdef __APPLE__` macros.

Reply:
I will try it then.

Reply:
> `MSG_NOSIGNAL` should be defined as [`SO_NOSIGPIPE`](https://lists.apple.com/archives/macnetworkprog/2002/Dec/msg00091.html).

Not correct -- `SO_NOSIGPIPE` is a socket option while `MSG_NOSIGNAL` is a flag for `send`.